anvill/python/anvill/binja/bnprogram.py

```diff
@@ -0,0 +1,275 @@
+
+            index += 1
+
+        ret_list = []
+        retTy = self.type_cache.get(bn_func.return_type)
+        if not isinstance(retTy, VoidType):
+            for reg in calling_conv.return_regs:
+                loc = Location()
+                loc.set_register(reg.upper())
+                loc.set_type(retTy)
+                ret_list.append(loc)
+
+        func = BNFunction(bn_func, arch, address, param_list, ret_list, func_type)
+        return func
+
+    def get_symbols_impl(self, address):
+        for s in self._bv.get_symbols(address, 1):
+            yield s.name
+
+    @property
+    def functions(self):
+        for f in self._bv.functions:
+            yield f.start
+
+    @property
+    def symbols(self):
+        for s in self._bv.get_symbols():
+            yield (s.address, s.name)
+
+    def _init_ctrl_flow_redirections(self):
+        """Initializes the control flow redirections using function thunks"""
+
+        # We only support the ELF format for now
+        # TODO
+
+        # List the function thunks first
+        input_file_path = self._bv.file.filename
+        image_parser = create_elf_image_parser(input_file_path)
+        function_thunk_list = image_parser.get_function_thunk_list()
+
+        # Go through each function thunk and add the redirection. Note that
+        # the __libc_start_main thunk does not need redirection since it's
+        # called directly without any wrapper function from the module entry
+        # point
+        is_32_bit = image_parser.get_image_bitness() == 32
+
+        reader = bn.BinaryReader(self._bv, bn.Endianness.LittleEndian)
+
+        redirected_thunk_list = []
+
+        for function_thunk in function_thunk_list:
+            # Read the call destination
+            reader.seek(function_thunk.rva)
+            redirection_dest = reader.read32() if is_32_bit else reader.read64()
+
+            # Get the variable defined at the dest address
+            func_location = self._bv.get_data_var_at(function_thunk.rva)
+            if not func_location:
+                print("anvill: No variable defined for {:x}".format(function_thunk.rva))
+                continue
+
+            # We should only have one caller
+            for caller in func_location.code_refs:
+                # Get the function containing this address; we need it to determine
+                # its start address
+                for caller_function in self._bv.get_functions_containing(
+                    caller.address
+                ):
+                    if (
+                        function_thunk.name == "__libc_start_main"
+                        or function_thunk.name not in caller_function.name
+                    ):
+                        continue
+
+                    redirection_source = caller_function.start
+
+                    print(
+                        "anvill: Redirecting thunk {:x}/{} user {:x} to {:x}".format(
+                            function_thunk.rva,
```

Comment:
I noticed this `.rva` use; elsewhere I thing we use things like `.start`; should we be switching to `.rva`?